### DISCUSSION TITLE
Memory Speed

### DISCUSSION THREAD

**Discussion by josh4789** [fastfetch-cli/fastfetch#1137] [Q&A]
I haven't dug that far down into the config but using `fastfetch -c all.jsonc` I dont see memory speed and I dont see it under `fastfetch -h memory-format`

I used `sudo dmidecode --t memory | grep Speed` to get my speed (and my configured memory speed dont know the difference yet and also dont know how to grab just one atm but ill figure that out if this seems simple enough) 

Im getting 
Speed: 5600 MT/s
Configured Memory Speed: 6000 MT/s
And I d'ont actually know which is the right one either

Is this something I could simply add myself in the config?

```
    "modules": [
        {
            "type": "command",
            "text": "dmidecode --t memory | grep -m 1 '^[[:blank:]]*Speed'",
            "key": "memoryspeed"
        }
    ]
}
```
tried putting that at the end of the generated module list and got nothing plus it has the downside of now needing sudo to even run because of dmidecode. Trying to find an alternative but this current one isnt working so i think im doing something wrong anyway

**Comment by CarterLi** [comment]
We have PhysicalMemory module

Try `sudo -E fastfetch -s physicalmemory`

**Reply by josh4789** [reply]
All that is outputting is the arch logo and no further output

**Reply by josh4789** [reply]
Also tried putting that into the module list in ~/.config/fastfetch and nothing happened there either as user or sudo
Im on arch using the extra repo not the -git from aur if that is important?

**Reply by CarterLi** [reply]
Try `sudo -E fastfetch -s physicalmemory --show-errors`

**Reply by josh4789** [reply]
Still just the arch logo

**Reply by CarterLi** [reply]
What fastfetch version are you using?

**Reply by josh4789** [reply]
2.20.0-1 thats what the repo says at least unless their is a command to use within fastfetch

**Reply by CarterLi** [reply]
Really strange. Can you please upload `/sys/firmware/dmi/tables/DMI` for debugging?

**Comment by josh4789** [comment]
[DMI.txt](https://github.com/user-attachments/files/16435429/DMI.txt)

Its a mess do they always look like that?

**Reply by CarterLi** [reply]
Strange.

What does `sudo dmidecode --t memory` print?

**Comment by josh4789** [comment]
[dmidecode.txt](https://github.com/user-attachments/files/16435650/dmidecode.txt)

**Comment by CarterLi** [comment]
Please test the latest dev build https://github.com/fastfetch-cli/fastfetch/commit/bd5012edc89f96450d3444abd9b390c9f858719b

`sudo -E` is still required

**Reply by josh4789** [reply]
`sudo -E fastfetch -s physicalmemory` does pull the memory modules size and speed now 

Is there something I can do in config to add that to the main command understanding it would then need sudo?
I tried adding in `"physicalmemory",` but nothing happens

**Reply by josh4789** [reply]
Oh i see `sudo -E fastfetch -c all` pulls it hmmm
